tracer/src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -109,11 +118,27 @@ public TracerSettings(IConfigurationSource source)
             // Filter out tags with empty keys or empty values, and trim whitespaces
             HeaderTags = InitializeHeaderTags(inputHeaderTags, headerTagsNormalizationFixEnabled);
 
-            var serviceNameMappings = source?.GetDictionary(ConfigurationKeys.ServiceNameMappings)
-                                            ?.Where(kvp => !string.IsNullOrWhiteSpace(kvp.Key) && !string.IsNullOrWhiteSpace(kvp.Value))
-                                            ?.ToDictionary(kvp => kvp.Key.Trim(), kvp => kvp.Value.Trim());
+            Dictionary<string, string> serviceNameMappings;
+            if (source?.GetDictionary(ConfigurationKeys.ServiceNameMappings) is { Count: > 0 } tmpServiceNameMappings)
+            {
+                var sNameMaps = new Dictionary<string, string>();
```

Comment:
initialize with capacity?

Reply:
done in [9b271d6](https://github.com/DataDog/dd-trace-dotnet/pull/3735/commits/9b271d6330489f7f37df13159a5f41abe6ef8bb8)